src/sentry/api/endpoints/project_details.py

```diff
@@ -488,6 +521,18 @@ def put(self, request: Request, project) -> Response:
         elif result.get("isBookmarked") is False:
             ProjectBookmark.objects.filter(project_id=project.id, user_id=request.user.id).delete()
 
+        if result.get("recapServerUrl") is not None:
+            if result["recapServerUrl"] == "":
+                project.delete_option(RECAP_SERVER_URL_OPTION)
+            elif project.get_option(RECAP_SERVER_URL_OPTION) != result["recapServerUrl"]:
+                project.update_option(RECAP_SERVER_URL_OPTION, result["recapServerUrl"])
+                poll_project_recap_server(project.id)
```

Comment:
Usually its an async task, because it might be slow. I believe this is a sync call here, so it will block the web request until it completes. Might be worth adding a `.delay` to it.

Reply:
Good catch